pkg/nvml/gpm.go

```diff
@@ -57,14 +57,23 @@ type GpmMetricsGetVType struct {
 func (l *library) GpmMetricsGetV(metricsGet *GpmMetricsGetType) GpmMetricsGetVType {
 	return GpmMetricsGetVType{metricsGet.convert()}
 }
+
+// nvmlGpmMetricsGetStub is a stub function that can be overridden for testing.
+var nvmlGpmMetricsGetStub = nvmlGpmMetricsGet
+
 func (metricsGetV GpmMetricsGetVType) V1() Return {
 	metricsGetV.metricsGet.Version = 1
-	return nvmlGpmMetricsGet(metricsGetV.metricsGet)
+	return nvmlGpmMetricsGetStub(metricsGetV.metricsGet)
 }
 
 func (l *library) GpmMetricsGet(metricsGet *GpmMetricsGetType) Return {
 	metricsGet.Version = GPM_METRICS_GET_VERSION
-	return nvmlGpmMetricsGet(metricsGet.convert())
+	nvmlMetricsGet := metricsGet.convert()
+	defer func() {
+		metricsGet.Metrics = nvmlMetricsGet.Metrics
+	}()
```

Comment:
Can we avoid doing a defer like this and do something more explicit?

Reply:
I remember now -- because the API requires some fields to be set as INPUTs and other fields to be set as OUTPUTs in the struct: https://github.com/NVIDIA/go-nvml/blob/main/gen/nvml/nvml.h#L9401-L9409